### DISCUSSION TITLE
Two questions about multi speaker models

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#175] [Archived]
<USER>>>> nmstoker<br/> <DATE> [April 3, 2020, 11:15pm]<br/> <br/> I realise the best answer is for me to simply try it, but thought I'd<br/> ask those who'd tried it for some general impressions first.<br/> <br/> **Q.** With multi speaker models in TTS, is there any discernible<br/> influence from one voice to another? For instance do they start to pick<br/> up characteristics from each other or do the voices remain distinct and<br/> like the original?<br/> <br/> **Q.** And do the voices yield better quality output than one would get<br/> by training just for one of the voices? By this, I mean is the act of<br/> training them all together helping the model learn overall common<br/> characteristics of language that it applies to all voices?<br/> <br/> I'm wondering about whether a multispeaker model with several relatively<br/> similar voices would help them get to a better quality because they<br/> might somehow reinforce each other or whether it's actually better to go<br/> for quite distinct voices (eg perhaps different accents or speaking<br/> patterns) which I'm guessing would generally be less likely to reinforce<br/> (if that happens at all?!) but may have some other advantage?<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/two-questions-about-multi-speaker-models]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [April 6, 2020, 8:55am]<br/> <br/> In my experience, training a multi-speaker model is more challanging<br/> since each speaker has a different distribution and different prosody.<br/> It makes things harder for the attention to align and decoder to learn<br/> the right voice.<br/> <br/> However, it's a different story for the vocoder model. It works better<br/> with multi-speaker training but it takes longer to converge.<br/> <br/> So for TTS model, they don't help each other to learn but they compete.<br/> Therefore, for the best quality, the dataset should be balanced in terms<br/> of number of records from each speaker.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [April 7, 2020, 10:49am]<br/> <br/> I have been working with Multispeaker for a while now but for the last<br/> two weeks it's been in the backlog because of things. My objective is<br/> more of figuring out if I actually **can** get a voice with infuences<br/> from a sum of n voices off the dataset, by feeding the TTS the mean of<br/> these speakers' embeddings. I actually started training a multispeaker<br/> TTS on LibriTTS with graves and I think I got it to 150k steps, where it<br/> was able to produce good speech, although it didn't by any means<br/> converge to all voices; it was able to produce speech in different<br/> voices if I gave it different ID's and, indeed, a random vector actually<br/> yielded a voice with mixed characteristics, but I think it was too soon<br/> that I stopped the training. What I noticed is that the first voice (id<br/> 0) was the most prominent one, at that point, although another voice was<br/> also there. Eren mentioned that, in general, a multispeaker TTS should<br/> be able to generalize okay, but I gather it greatly depends on the<br/> dataset and the voice representation stats, in order to get a good<br/> vector space representation. It also depends on your use case -- if you<br/> want a single speaker TTS that is of good quality, I would guess it<br/> might be better to finetune a pretrained model (the best one for me so<br/> far has been the one trained with ForwardAttn and BN norm). As far as<br/> external embeddings go, I have tried to do some work<br/> [here](https://github.com/george-roussos/TTS) but it has failed some<br/> Travis tests so Eren was not able to merge it and I have not yet found<br/> time to look into what is wrong.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [April 7, 2020, 12:41pm]<br/> <br/> Thank you both for sharing those insights, it's very much appreciated.<br/> <br/> Looks like I'll have to start weighing up options regarding datasets and<br/> try a few. I'm not *quite* there yet as am still (slowly!) working on<br/> methods to better handle pronunciation in my main dataset (to spot cases<br/> where the phonemes aren't accurate or I've got heteronyms)<br/> <br/> Will let you know how I get on when I do try this out. Thanks again!<br/> <br/> [Archived Post]<br/>
